### ISSUE TITLE
Nullness - Consider supporting Nullable shorthand for TreatWarningsAsErrors

### ISSUE THREAD

**Issue by BoundedChenn31** [dotnet/fsharp#17524]
**Is your feature request related to a problem? Please describe.**

In C# it's possible to convert all nullable-related warnings to errors using `<WarningsAsErrors>nullable</WarningsAsErrors>`. From [docs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/errors-warnings#treatwarningsaserrors):
`
The set of all nullability warnings can be specified with the Nullable shorthand.
`

So in line with https://github.com/dotnet/fsharp/pull/17486 it would be nice to have WAE behaviour as well.

**Describe the solution you'd like**

Support `<WarningsAsErrors>nullable</WarningsAsErrors>`

**Describe alternatives you've considered**

Just use specific warning numbers or even `true`.

**Additional context**

.

**Comment by T-Gro** [comment]
So the `nullable` would be a shorthand for putting in all nullness-related errors, on top of the option for specifying error codes individually.

Do I understand it right?

**Comment by vzarytovskii** [comment]
> So the `nullable` would be a shorthand for putting in all nullness-related errors, on top of the option for specifying error codes individually.
> 
> Do I understand it right?

Yeah, I _think_ that's what C# does currently.

**Comment by BoundedChenn31** [comment]
> So the `nullable` would be a shorthand for putting in all nullness-related errors, on top of the option for specifying error codes individually.
> 
> Do I understand it right?

Yes, not sure about implementation details (csc also supports `/warnaserror:nullable` option, so maybe it's handled completely by roslyn) but it definitely behaves like that and recommended as a solution:

SO: [How to treat ALL C# 8 nullable reference warnings as errors?](https://stackoverflow.com/questions/58194983/how-to-treat-all-c-sharp-8-nullable-reference-warnings-as-errors)
Roslyn 1: [Provide single option to treat nullable issues as errors not warnings](https://github.com/dotnet/roslyn/issues/41605)
Roslyn 2: [Proposal: Add switch for treating nullable-related warnings as errors](https://github.com/dotnet/roslyn/issues/44721)

**Comment by abonie** [comment]
It's done, reopen if you still see it with F# 9

**Comment by T-Gro** [comment]
This was not done, this proposal is about a syntactical shorthand for doing warningsAsErrors for all nullable-related warnings. (as opposed to listing individual error codes)

**Comment by vzarytovskii** [comment]
I thought that we did it already for some reason
